consensus/src/liveness/proposal_generator_test.rs

```diff
@@ -108,16 +134,58 @@ async fn test_old_proposal_generation() {
         Arc::new(MockPayloadManager::new(None)),
         Arc::new(SimulatedTimeService::new()),
         1,
+        10,
     );
+    let mut proposer_election = UnequivocalProposerElection::new(Box::new(RotatingProposer::new(
+        vec![inserter.signer().author()],
+        1,
+    )));
     let genesis = block_store.ordered_root();
     let a1 = inserter
         .insert_block_with_qc(certificate_for_genesis(), &genesis, 1)
         .await;
     inserter.insert_qc_for_block(a1.as_ref(), None);
 
     let proposal_err = proposal_generator
-        .generate_proposal(1, empty_callback())
+        .generate_proposal(1, &mut proposer_election, empty_callback())
         .await
         .err();
     assert!(proposal_err.is_some());
 }
+
+#[tokio::test]
+async fn test_correct_failed_authors() {
```

Comment:
Can we add some negative test cases as well - ensuring we fail when failed authors is not well formed?